File: src/Desafio.Domain/Entidades/Marca.cs
using Desafio.Domain.Enums;
using Desafio.Domain.ValueObject;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desafio.Domain.Entidades
{
    public class Marca : Entity
    {
        public Nome Nome { get; private set; }
        public EStatus Status { get; private set; }
        private readonly List<Veiculo> _veiculos;
        public IReadOnlyCollection<Veiculo> Veiculos => _veiculos;

        public Marca(Nome nome, EStatus status)
        {
            Nome = nome;
            _veiculos = new List<Veiculo>();
            Status = status;
        }

        protected Marca()
        {

        }

        // QUESTIONAR SOBRE REGRAS DE NEGÓCIOS

        //public void AlterarNome(string nome)
        //{
        //    if(string.IsNullOrEmpty(nome))
        //    {
        //        Nome.Valor = nome;
        //    }
        //}

        public void Ativar()
        {
            Status = EStatus.Ativo;
        }

        public void Cancelar()
        {
            Status = EStatus.Cancelado;
        }

    }
}

File: src/Desafio-FrontEnd/src/app/models/marca.ts
import { Nome } from "./nome";
import { Veiculo } from "./veiculo";

export class Marca {

    constructor(
        id?: string,
    nome?: Nome,
    status?: number,
    ) {
        this.id = id
        this.nome = nome,
        this.status = status
        this. veiculos = []
    }

    public id?: string;
    public nome?: Nome;
    public status?: number;
    public veiculos?: Veiculo[]
}